File: tools/bench/answers/python_sonnet/s1/09_topwords.py
sentence = "the cat sat on the mat with the cat"
words = sentence.split()
counts = {}
for w in words:
    counts[w] = counts.get(w, 0) + 1
sorted_words = sorted(counts.items(), key=lambda x: x[1], reverse=True)
for word, count in sorted_words[:2]:
    print(word, count)

File: tools/bench/answers/rust_haiku/s2/09_topwords.rs
fn main() {
    let sentence = "the cat sat on the mat with the cat";
    let words: Vec<&str> = sentence.split_whitespace().collect();

    let mut counts = Vec::new();

    for word in &words {
        let mut found = false;
        for (w, c) in &mut counts {
            if w == word {
                *c += 1;
                found = true;
                break;
            }
        }
        if !found {
            counts.push((word, 1));
        }
    }

    counts.sort_by_key(|&(_, c)| -(c as i32));

    for i in 0..2 {
        let (word, count) = counts[i];
        println!("{} {}", word, count);
    }
}
